tests/api/test_topic_entity_tag.py

```diff
@@ -0,0 +1,218 @@
+            assert tet_obj.entity_type == "Gene"
+            assert tet_obj.alliance_entity == "Bob_gene_name"
+            assert tet_obj.taxon == "NCBITaxon:1234"
+            assert tet_obj.note == "Some Note"
+
+            props = db.query(TopicEntityTagPropModel).filter(
+                TopicEntityTagPropModel.topic_entity_tag_id == test_topic_entity_tag.new_tet_id).all()
+
+            count = 0
+            for prop in props:
+                if prop.qualifier in ["Quali1", "Quali2"]:
+                    count += 1
+                else:
+                    assert "Diff qualifier" == prop.qualifier
+            assert count == 2
+
+            response = client.get(url=f"/topic_entity_tag/{test_topic_entity_tag.new_tet_id}")
+            res = response.json()
+            assert res["topic"] == "Topic1"
+            assert res["props"][0]["qualifier"] == "Quali1"
+            assert res["props"][1]["qualifier"] == "Quali2"
+
+    def test_create_bad(self, test_reference, auth_headers): # noqa
+        with TestClient(app) as client:
+            xml = {
+                "reference_curie": test_reference.json(),
+                "topic": "Topic1",
+                "entity_type": "Gene",
+                "taxon": "NCBITaxon:1234"
+            }
+            # No Entities
+            response = client.post(url="/topic_entity_tag/", json=xml, headers=auth_headers)
+            assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
+
+            # More than one Entity
+            xml2 = copy.deepcopy(xml)
+            xml2["alliance_entity"] = "Bob_gene_name 1"
+            xml2["mod_entity"] = "Bob_gene_name 2"
+            response = client.post(url="/topic_entity_tag/", json=xml2, headers=auth_headers)
+            assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
+
+            # No curie
+            xml3 = copy.deepcopy(xml2)
+            del xml3["mod_entity"]
+            del xml3["reference_curie"]
+            response = client.post(url="/topic_entity_tag/", json=xml3, headers=auth_headers)
+            assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
+
+            # Bad curie
+            xml4 = copy.deepcopy(xml3)
+            xml4["reference_curie"] = "BADCURIE"
+            response = client.post(url="/topic_entity_tag/", json=xml4, headers=auth_headers)
+            assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
+
+            # No species
+            xml5 = copy.deepcopy(xml4)
+            del xml5["taxon"]
+            xml5["reference_curie"] = test_reference.json()
+            response = client.post(url="/topic_entity_tag/", json=xml4, headers=auth_headers)
+            assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
+
+    def test_patch_with_props(self, test_topic_entity_tag, test_reference, auth_headers): # noqa
+        with TestClient(app) as client:
+            # change the reference
```

Comment:
I'm not sure this is testing changing the reference, is the reference_curie being patched the same as the original and then testing that the response is still the original/patch ?

Reply:
I'm not sure about this. I'll take a look and modify the test if needed. I tried to keep the same logic as the original unit test file but maybe something got lost here

Reply:
we fixed this issue together by using test_reference2